src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -56,10 +57,34 @@ public final class QueryInsightsListener extends SearchRequestOperationsListener
      */
     @Inject
     public QueryInsightsListener(final ClusterService clusterService, final QueryInsightsService queryInsightsService) {
+        super(false); // Disable query insights listener and service initially
         this.clusterService = clusterService;
         this.queryInsightsService = queryInsightsService;
-        // Setting endpoints set up for top n queries, including enabling top n queries, window size and top n size
-        // Expected metricTypes are Latency, CPU and Memory.
+        initialize();
+    }
+
+    /**
+     * Constructor for QueryInsightsListener
+     *
+     * @param clusterService       The Node's cluster service.
+     * @param queryInsightsService The topQueriesByLatencyService associated with this listener
+     * @param initiallyEnabled Is the listener initially enabled/disabled
+     */
+    public QueryInsightsListener(
+        final ClusterService clusterService,
+        final QueryInsightsService queryInsightsService,
+        boolean initiallyEnabled
+    ) {
+        super(initiallyEnabled);
+        this.clusterService = clusterService;
```

Comment:
Why do we need to store the clusterService reference?

Reply:
Need in `constructSearchQueryRecord` method